alsaaudio.c

```diff
@@ -572,6 +630,169 @@ alsapcm_dumpinfo(alsapcm_t *self, PyObject *args)
     return Py_None;
 }
 
+// auxiliary function
+
+
+static PyObject *
+alsapcm_getformats(alsapcm_t *self, PyObject *args) {
+	snd_pcm_t *pcm = self->handle;
+		if (!pcm) {
```

Comment:
Inconsistent whitespace?

Reply:
Yes - my Eclipse editor was having a very strange day, an the indentations were all over the place.  Sorry.  Can fix if required.

I'll have a think about tests - probably can't just be pure Python.